rj_param_utils/src/global_param.cpp

```diff
@@ -0,0 +1,23 @@
+#include <rj_param_utils/global_param.hpp>
+
+ROS2GlobalParameterProvider::ROS2GlobalParamProvider(rclcpp::Node* node, 
+        const std::string& global_node) : node_{node}, global_node_{global_node} {
+    params_client_ = std::make_shared<rclcpp::AsyncParametersClient>(node);
+
+    auto on_param_event_callback = 
+        [this](const rcl_interfaces::msg::ParameterEvent::SharedPtr event) -> void
+        {
+            std::vector<std::string> names;
+            for (auto& changed_parameter : event->changed_parameters) {
+                names.emplace_back(changed_parameter.name);
+                Update(changed_parameter.name, params_client_->get_parameters(names));
```

Comment:
I think the parameter values should be contained in the `changed_parameter` field, as `changed_parameter.value`. It is of type `ParameterValue`:

```
# Used to determine which of the next *_value fields are set.
# ParameterType.PARAMETER_NOT_SET indicates that the parameter was not set
# (if gotten) or is uninitialized.
# Values are enumerated in `ParameterType.msg`.

# The type of this parameter, which corresponds to the appropriate field below.
uint8 type

# "Variant" style storage of the parameter value. Only the value corresponding
# the type field will have valid information.

# Boolean value, can be either true or false.
bool bool_value

# Integer value ranging from -9,223,372,036,854,775,808 to
# 9,223,372,036,854,775,807.
int64 integer_value

# A double precision floating point value following IEEE 754.
float64 double_value

# A textual value with no practical length limit.
string string_value

# An array of bytes, used for non-textual information.
byte[] byte_array_value

# An array of boolean values.
bool[] bool_array_value

# An array of 64-bit integer values.
int64[] integer_array_value

# An array of 64-bit floating point values.
float64[] double_array_value

# An array of string values.
string[] string_array_value
```
However, IIRC this type is not accepted by `Update` directly. I would break out this lambda:
https://github.com/RoboJackets/robocup-software/blob/1e65413efbcef57f50f516279854f22f1c2334f0/rj_param_utils/src/ros2_param_provider.cpp#L69-L115
into a separate `protected` method in `ROS2ParamProvider`, and then call it here.